src/geode/mesh/core/grid.cpp

```diff
@@ -324,25 +268,27 @@ namespace geode
                     direction, "." );
             }
             double nb_vertices_double{ 1 };
-            for( const auto c : cells_number_ )
+            for( const auto d : LRange{ dimension } )
             {
-                nb_vertices_double *= static_cast< double >( c + 1 );
+                nb_vertices_double *= static_cast< double >(
+                    grid.nb_cells_in_direction( d ) + 1 );
             }
             OPENGEODE_EXCEPTION(
                 nb_vertices_double < static_cast< double >( UINT_MAX ),
                 "[Grid] Creation of a grid for which the number of cell "
                 "vertices exceeds the unsigned int limit." );
-            OPENGEODE_EXCEPTION( nb_cells() != 0,
-                "[Grid] Creation of a grid with no cells "
-                "in one direction." );
         }
 
         void copy( const Grid< dimension >::Impl& impl )
         {
-            cells_number_ = impl.cells_number_;
             cells_length_ = impl.cells_length_;
         }
 
+        std::array< index_t, dimension >&& deprecated_cells_number()

```

Comment:
I did not find a better way.. I have to leave it there so the Grid class can copy it into the Array..

Reply:
I do not get that.
why do you need that in the array?
deprecated cells only apperas in structural grid and a structural grid cannot be represented in a simple aray. am i wrong?

Reply:
We need it for the serialize function, allowing the loading of previous versions of the grid class.